xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAESink.cpp

```diff
@@ -728,6 +728,111 @@ void CActiveAESink::PrintSinks(std::string& driver)
   }
 }
 
+std::string CActiveAESink::ValidateOuputDevice(const std::string& device)
+{
+  if (m_sinkInfoList.empty())
+    return {};
+
+  const AESinkDevice dev = CAESinkFactory::ParseDevice(device);
+
+  // find exact match of deviceName in same driver
+  if (!dev.driver.empty() && !dev.name.empty())
+  {
+    for (const auto& sink : m_sinkInfoList)
+    {
+      if (sink.m_sinkName != dev.driver)
+        continue;
+
+      for (const auto& d : sink.m_deviceInfoList)
+      {
+        if (d.m_deviceName == dev.name)
+        {
+          std::string device = dev.driver + ":" + dev.name;
+
+          // note that friendly name may have changed, then is used
+          // current "physical" name (not name saved in settings)
+          // in this way friendly name is updated over time (drivers updates, etc.)
+          const std::string friendlyName = d.GetFriendlyName();
+          if (!friendlyName.empty())
+          {
+            device.append(":");
+            device.append(friendlyName);
+          }
+
+          return device;
```

Comment:
Since the purpose of this function is to find best enumerated device, I think it would make sense to create a  member function CAEDeviceInfo::ToDeviceString() that returns a formatted device string instead of the 4x repetition of the string building code in this function.

Reply:
Implemented